pin/pin.go

```diff
@@ -221,30 +231,51 @@ func (p *pinner) RemovePinWithMode(key util.Key, mode PinMode) {
 func LoadPinner(d ds.ThreadSafeDatastore, dserv mdag.DAGService) (Pinner, error) {
 	p := new(pinner)
 
+	rootKeyI, err := d.Get(pinDatastoreKey)
```

Comment:
`/local/pins/{direct,recursive,indirect}/keys` (old way) and `/local/pins` (new way) are keys in the `Datastore`, not really IPFS objects (yet). No Link-following there.

I would rather get things mainlined and even potentially inconvenience users than have a branch stay alive for a month or two, accumulating merge conflicts. But that's just me.

All the new pinning code needs is a place to store a single root object hash. At the point the `object-repo` stuff lands, it'll have to convert things from datastore to being stored inside ipfs objects anyway -- and this (datastore key `/local/pins` -> fields Pins in the root object) will just be a part of that migration. It doesn't really change the risk of that migration, in my mind.

Squirrel says ship it.